### PR TITLE
Add support for internal copy in S3 blob db

### PR DESCRIPTION
**Pull request by millerdev**
Internal copy performs an efficient copy operation when `put`ting a blob that was previously loaded using `get`.

A new version of boto3 (v1.4.0) is needed for the copy operation. The upgrade also simplified multipart upload support.

@snopoke cc @NoahCarnahan

### CODE REVIEW

File: corehq/blobs/s3db.py [hunk 1]
```diff
@@ -8,7 +9,7 @@
 from corehq.blobs.util import ClosingContextProxy
 
 import boto3
-from boto3.s3.transfer import S3Transfer, ReadFileChunk
+from boto3.s3.transfer import S3Transfer
```
Comment (dimagimon) [inline corehq/blobs/s3db.py]: F401 'S3Transfer' imported but unused

File: corehq/blobs/s3db.py [hunk 2]
```diff
@@ -134,149 +160,3 @@ def maybe_not_found(throw=None):
...
-                    self.file_locks.pop(self.file)
-        finally:
-            self.file = None
-            self.lock = None
-
-    def __enter__(self):
-        return self
-
-    def __exit__(self, *args, **kwargs):
-        self.close()
```
Comment (snopoke) [inline corehq/blobs/s3db.py]: :skull:

**Comment by snopoke** [comment]
Nice

File: corehq/blobs/s3db.py [hunk 3]
```diff
@@ -78,9 +82,8 @@ def delete(self, *args, **kw):
     def copy_blob(self, content, info, bucket):
         self._s3_bucket(create=True)
         path = self.get_path(info.identifier, bucket)
-        osutil = OpenFileOSUtils()
-        transfer = S3Transfer(self.db.meta.client, osutil=osutil)
-        transfer.upload_file(content, self.s3_bucket_name, path)
+        self._s3_bucket().upload_fileobj(content, path)
```
Comment (millerdev) [inline corehq/blobs/s3db.py]: Notable difference in new `upload_fileobj` implementation in boto3 (v1.4.0): https://github.com/boto/s3transfer/issues/52

**Pull request merged**